client/lib/analytics/utils/refresh-country-code-cookie-gdpr.js

```diff
@@ -20,35 +20,34 @@ export default async function refreshCountryCodeCookieGdpr( signal = undefined )
 	}
 
 	if ( refreshRequest === null ) {
-		refreshRequest = requestGeoData( signal ).then( ( { countryCode, region } ) => {
-			setCountryCodeCookie( countryCode );
-			setRegionCookie( region );
-		} );
+		refreshRequest = requestGeoData( signal )
```

Comment:
One of the concerns I had from https://github.com/Automattic/wp-calypso/pull/102862#issuecomment-2855494502 (the "compensating behavior" remark) was that we'd previously wait 6 hours between request attempts due to the cookie duration, even if the request failed. Now we might be sending requests much more frequently, and in the case of an ad blocker, they would continuously fail.

Should we have anything to try to limit that, or do you think it's unlikely to be a problem?

Reply:
Well, maybe we should set the `unknown` cookie values after all 🙂 The caveat is that we shouldn't _overwrite_ an existing cookie when the `/geo` request fails: only set it to `unknown` when it doesn't exist. That will solve both problems: i) good data is not overwritten with bad data; ii) the `unknown` value will signal that the `/geo` request was already sent in the last 6 hours, no point re-requesting until the cookie expires.